src/span.js

```diff
@@ -154,11 +160,12 @@ export default class Span {
      * @param {number} finishTime - The time on which this span finished.
      **/
     finish(finishTime: ?number): void {
-        if (this._duration !== undefined) {
+        if (this.finished) {
             throw new Error('You can only call finish() on a span once.');
```

Comment:
how about we just log it? not only throwing exception breaks the 1st rule of monitoring "do no harm", but it also prevents optimization of the method by the JS engine.